src/Arcus.WebApi.Unit/Correlation/TelemetryCorrelationTests.cs

```diff
@@ -0,0 +1,114 @@
+                             OperationIdPropertyName = "OperationId";
+
+        private readonly TestApiServer _testServer = new TestApiServer();
+
+        [Fact]
+        public async Task SendRequest_WithSerilogCorrelationEnrichment_ReturnsOkWithEnrichedCorrelationLogProperties()
+        {
+            // Arrange
+            using (HttpClient client = _testServer.CreateClient())
+            // Act
+            using (HttpResponseMessage response = await client.GetAsync(Route))
+            {
+                // Assert
+                Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                CorrelationInfo correlationInfo = await AssertAppCorrelationInfoAsync(response);
+                AssertLoggedCorrelationProperties(correlationInfo);
+            }
+        }
+
+        [Fact]
+        public async Task SendRequest_WithSerilogCorrelationenrichment_ReturnsOkWithDifferentOperationIdAndSameTransactionId()
+        {
+            // Arrange
+            using (HttpClient client = _testServer.CreateClient())
+            // Act
+            using (HttpResponseMessage firstResponse = await client.GetAsync(Route))
+            {
+                // Assert
+                Assert.Equal(HttpStatusCode.OK, firstResponse.StatusCode);
+                CorrelationInfo firstCorrelationInfo = await AssertAppCorrelationInfoAsync(firstResponse);
+                AssertLoggedCorrelationProperties(firstCorrelationInfo);
+
+                var request = new HttpRequestMessage(HttpMethod.Get, Route);
+                request.Headers.Add("X-Transaction-ID", firstCorrelationInfo.TransactionId);
+                
+                using (HttpResponseMessage secondResponse = await client.SendAsync(request))
+                {
+                    Assert.Equal(HttpStatusCode.OK, secondResponse.StatusCode);
+                    CorrelationInfo secondCorrelationInfo = await AssertAppCorrelationInfoAsync(secondResponse);
+                    AssertLoggedCorrelationProperties(secondCorrelationInfo);
+
+                    Assert.NotEqual(firstCorrelationInfo.OperationId, secondCorrelationInfo.OperationId);
+                    Assert.Equal(firstCorrelationInfo.TransactionId, secondCorrelationInfo.TransactionId);
+                }
+            }
+        }
+
+        [Fact]
+        public async Task SendRequest_WithSerilogCorrelationenrichment_ReturnsOkWithDifferentOperationIdAndDifferentTransactionId()
+        {
+            // Arrange
+            using (HttpClient client = _testServer.CreateClient())
+            // Act
+            using (HttpResponseMessage firstResponse = await client.GetAsync(Route))
+            {
+                // Assert
+                Assert.Equal(HttpStatusCode.OK, firstResponse.StatusCode);
+                CorrelationInfo firstCorrelationInfo = await AssertAppCorrelationInfoAsync(firstResponse);
+                AssertLoggedCorrelationProperties(firstCorrelationInfo);
```

Comment:
We should pass a transaction id in the request and also verify if that matches what was found

Reply:
Isn't that what happens in the test above this one?

Reply:
Ah my bad, sorry - Looks like I'm having a bad day :/